src/converters.ts

```diff
@@ -216,21 +256,27 @@ function decimal(options?: DecimalOptions) {
       raw = raw.trim();
       return convertSeparators(raw, options);
     },
-    rawValidate(raw) {
+    rawValidate(raw, options) {
       if (raw === "" || raw === ".") {
         return false;
       }
+      if (
```

Comment:
I don't understand this code. This means that if you set options.thousandSeparator and no decimalSeparator, the raw value is always rejected as invalid. I think this is intended to forbid this combination but this doesn't do that -- it allows this combination but then always declares what the user enters as invalid. I expect some code somewhere (earlier than this, for instance when you pass these options into state) that throws an error for this illegal combination.

Reply:
And a test for it, of course.

Reply:
I added a comment later on for the test that I want to refer to here: you *could* implement a throw here when this impossible situation is reached. This is a safeguard that shouldn't be reached in real-world code, as it's later than we should make it (it only occurs once the user starts entering data). Instead, we can catch this earlier -- the earliest place is when we pass the options into `state()`.